src/lib/validator.ts

```diff
@@ -15,6 +15,32 @@ export const hasValue = (val: undefined | null | string): val is string => {
   return val !== undefined && val !== null && val !== "";
 };
 
+/**
+ * Returns true of val is not undefined, not null and not an empty string
+ * and can be converted to integer.
+ *
+ * @param val Value to inspect
+ */
+export const isInteger = (val: undefined | null | string): val is string => {
```

Comment:
when implementing type-guards, loosen the type constraint on `val` to be `unknown`:

```ts
 export const isInteger = (val: unknown): val is string => {
```

Reply:
also, this functions name is ambiguous. `isInteger` should correspond to the logic `Number.isInteger`. that is `type isInteger = (value: unknown) => value is number`.

The logic you've implemented should be named more aptly something like `isIntegerString()`

Reply:
I have no problem changing the function name if you really think that it is necessary. Please let me know

Reply:
I do not see that reason for loosing the type because we precisely know the types from commander is doing to be either undefined, null or string.

Reply:
- rename the function.
- we want to loosen the constraint on the function because you have written a general validation function that we should want to be able to reuse in other situations if need be (eg, if we want to use the function to parse values form config files or alike).

Reply:
accept